server-timing/class-perflab-server-timing-metric.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Server-Timing API: Perflab_Server_Timing_Metric class
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Class representing a single Server-Timing metric.
+ *
+ * @since n.e.x.t
+ */
+class Perflab_Server_Timing_Metric {
+
+	/**
+	 * The metric slug.
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	private $slug;
+
+	/**
+	 * The metric value in milliseconds.
+	 *
+	 * @since n.e.x.t
+	 * @var int|float|null
+	 */
+	private $value;
+
+	/**
+	 * The value measured before relevant execution logic in seconds, if used.
+	 *
+	 * @since n.e.x.t
+	 * @var float|null
+	 */
+	private $before_value;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $slug The metric slug.
+	 */
+	public function __construct( $slug ) {
+		$this->slug = $slug;
+	}
+
+	/**
+	 * Gets the metric slug.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string The metric slug.
+	 */
+	public function get_slug() {
+		return $this->slug;
+	}
+
+	/**
+	 * Sets the metric value.
+	 *
+	 * Alternatively to setting the metric value directly, the {@see Perflab_Server_Timing_Metric::measure_before()}
+	 * and {@see Perflab_Server_Timing_Metric::measure_after()} methods can be used to further simplify measuring.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param int|float $value The metric value to set, in milliseconds.
+	 */
+	public function set_value( $value ) {
+		if ( ! is_int( $value ) && ! is_float( $value ) ) {
```

Comment:
PHP is loosely typed language and it's pretty common situation when some numbers come as strings. So, I think it will just add more problems than benefits because people won't expect strict types checking here.

Reply:
@eugene-manuilov Fair point. I've updated this in https://github.com/WordPress/performance/pull/553/commits/569637dcca8249a7c2dd9228905bb5e3eaf764b4.

Reply:
Thanks, @felixarntz. Looks good to me.